src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge29.java

```diff
@@ -0,0 +1,89 @@
+package org.owasp.wrongsecrets.challenges.docker;
+
+
+import org.owasp.wrongsecrets.RuntimeEnvironment;
+import org.owasp.wrongsecrets.ScoreCard;
+import org.owasp.wrongsecrets.challenges.Challenge;
+import org.owasp.wrongsecrets.challenges.ChallengeTechnology;
+import org.owasp.wrongsecrets.challenges.Spoiler;
+import org.spongycastle.util.encoders.Hex;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import javax.crypto.Cipher;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.List;
+
+import static org.owasp.wrongsecrets.RuntimeEnvironment.Environment.DOCKER;
+
+@Component
```

Comment:
The suggestion below adds SL4jf logging, so you can do things like you can see in other classes when you catch an exception with "            log.warn("Exception when decrypting: {}", e.getMessage());"
```suggestion
@Component
@Slf4j
```